_pages/dev/security/security-headers.md

```diff
@@ -0,0 +1,139 @@
+**TODO**: WIP, discussion with security team and ccv2 ongoing.
+
+HTTP security headers are a important part of website security. They protect your storefront against potential attacks, such as XSS, code injection, clickjacking, etc.
+
+This document describes various security headers that are in place or should be added by customers when they deploy a Spartacus based Storefront in production.
+
+[Owasp](https://owasp.org/www-project-secure-headers/) gives a good overview of the various headers that can be applied. We've used the headers outlined by Owasp and reflect the recommended headers for Spartacus.
+
+## HTTP Strict-Transport-Security
+
+The HSTS security header forces the web browser to access the storefront by https only. This contributes to protocol downgrading and cookie hijacking.
+
+This is a pretty basic header that should be applied by default to Spartacus applications.
+
+The following snippets shows an example header config:
+
+```
+Strict-Transport-Security: max-age=31536000 ; includeSubDomains
+```
+
+**TODO**
+
+- @Michael Verify if this is added in ccv2 by default (likely already on ingress controller)
+
+## HTTP Public Key Pinning (HPKP)
+
+`HPKP` is a (deprecated) mechanism that can be added through http headers. The mechanism would forces the same headers cross various layers, including APIs.
+
+CCv2 doesn't support this header deliberately. It adds too much pressure on syncing certificates cross multiple (changing) layers in the hosting setup. Moreover, HPKP is deprecated and not implemented in a stable manner cross browsers for HPKP.
+
+Spartacus does not insist on the usage of HPKP.
+
+## X-Frame-Options
+
+The `X-Frame-Options` header is a well-known header that can be used to mitigate clickjacking. This header can only be 2 values, `DENY` or `SAMEORIGIN`. This is too limited for Spartacus as a default strategy to prevent clickjacking, since the Spartacus storefront runs in an iframe in SmartEdit. SmartEdit loads Spartacus in an iframe and actually uses _Clickjacking_ technique deliberately to support inline content managing.
+
+The `Content-Security-Policy: frame-ancestors` offers more sophisticated approach to tackle clickjacking, so that SmartEdit can be used while still being able to protect from clickjacking. The CSP is detailed further below.
+
+The `Content-Security-Policy: frame-ancestors` is not supported in older browsers such as IE11. Customers who need to support IE11, the `X-Frame-Options` could be used as an alternative policy to evaluate. (Keep in mind however that this will block SmartEdit on the same deployed environment).
```

Comment:
Not sure if we need to explain this better:
Modern browsers will ignore the `X-Frame-Options` header completely, if a CSP header with the `frame-ancesters` directive is also present.
(Says so in the standard, and I've also tested it successfully with several browsers.)

Reply:
That's great input actually. I therefor recommend to have both `X-Frame-Options` AND the CSP `frame-ancestors` directive.